### PR TITLE
Print Names of the Nodes being saved per chunk on Client at TRACE

### PR DESCRIPTION
**Pull request by sagarsane**
This will help #70  and also #66 a little bit. The logging is only enabled at TRACE for `batch-client.log` so it shouldn't harm much. The mandatory child nodes are currently logged as "relative" paths to their parents.

Example of logging : 

```
02.09.2015 22:27:59.284 *TRACE* [clientJobLauncherTaskExecutor-1] 
com.twcable.grabbit.client.batch.steps.jcrnodes.JcrNodesWriter Saving Nodes : 
[/content/dam/geometrixx-gov/resources-icon.png/jcr:content/renditions/cq5dam.thumbnail.48.48.png, 
/jcr:content, /content/dam/geometrixx-gov/resources-
icon.png/jcr:content/renditions/cq5dam.web.1280.1280.png, /jcr:content, /content/dam/geometrixx-
gov/resources-icon.png/jcr:content/renditions/cqdam.pyramid.tiff, /jcr:content, /content/dam/geometrixx-
gov/resources-icon.png/jcr:content/renditions/original, /jcr:content, /content/dam/geometrixx-
gov/logo.png, /content/dam/geometrixx-gov/logo.png/jcr:content, /content/dam/geometrixx-
gov/logo.png/jcr:content/metadata, /content/dam/geometrixx-
gov/logo.png/jcr:content/metadata/xmpMM:DerivedFrom, /content/dam/geometrixx-
gov/logo.png/jcr:content/related, /content/dam/geometrixx-gov/logo.png/jcr:content/renditions, /content/dam/geometrixx-gov/logo.png/jcr:content/renditions/cq5dam.thumbnail.140.100.png, /jcr:content, 
/content/dam/geometrixx-gov/logo.png/jcr:content/renditions/cq5dam.thumbnail.319.319.png, /jcr:content,
 /content/dam/geometrixx-gov/logo.png/jcr:content/renditions/cq5dam.thumbnail.48.48.png, /jcr:content,
 /content/dam/geometrixx-gov/logo.png/jcr:content/renditions/cq5dam.web.1280.1280.png, /jcr:content, 
/content/dam/geometrixx-gov/logo.png/jcr:content/renditions/cqdam.pyramid.tiff, /jcr:content, 
/content/dam/geometrixx-gov/logo.png/jcr:content/renditions/original, /jcr:content]
```

@jdigger 

cc : @jbornemann @jazeren1

### CODE REVIEW

File: grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy [hunk 1]
```diff
@@ -47,6 +47,9 @@ class JcrNodesWriter implements ItemWriter<ProtoNode>, ItemWriteListener {
     @Override
     void afterWrite(List nodeProtos) {
         log.info "Saving ${nodeProtos.size()} nodes"
+        log.trace """Saving Nodes : ${(nodeProtos as List<ProtoNode>).collectMany { ProtoNode pNode ->
+            [ pNode.name , pNode.mandatoryChildNodeList.collect { it.name - pNode.name }]
+        }.flatten()}"""
```
Comment (sagarsane) [inline grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy]: I can take out the `flatten()` or `it.name - pNode.name` if things are more clearer that way....

Comment (jdigger) [inline grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy]: Isn't `.flatten()` redundant with `.collectMany()`?

Go ahead and remove the `it.name - pNode.name` to make it clearer.

Comment (sagarsane) [inline grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy]: I think I was seeing the output like `[a, b, [b1, b2], c]` without the `flatten()` but I will double check and remove if it's not required.

**Comment by jdigger** [comment]
I'm thinking it would probably be better to log at DEBUG instead of TRACE. That's the level people are used to turning on if they want to see more than INFO. (log4j doesn't have TRACE, for example, so there's a lot of convention/history working against TRACE.) And we're keeping the logging "reasonably" light (one line per chunk, as opposed to one per node) so it's not the crazy amount that is usually associated with TRACE.

**Comment by jdigger** [comment]
In the README it should mention that if you want to see what nodes are being written then change the client LogFactory config node to DEBUG/TRACE

File: grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy (hunk 1 above)
Comment (jdigger) [inline grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy]: Come to think of it, that makes sense.  `.collectMany()` only does the "flatten" to one level, whereas `.flatten()` is recursive.

**Comment by sagarsane** [comment]
Okay. Updated the logging to be at `DEBUG` level. Also, added in the `README` about turning on DEBUG logs in my other PR #67

**Pull request merged**